mbed_build/_internal/config/config_modifiers/set_target_attribute.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Modify values in Config.target.*."""
+import itertools
+from dataclasses import dataclass
+from typing import Any, Callable, List, cast
+from typing_extensions import Literal
+
+# Fix circular dependency problem
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mbed_build._internal.config.config import Config
+
+
+class InvalidModifierData(Exception):
+    """Raised when data given to builder is not valid for target attribute modifier."""
+
+
+ACCUMULATING_OVERRIDES = ("extra_labels", "macros", "device_has", "features", "components")
+MODIFIERS = ("add", "remove")
+ALL_ACCUMULATING_OVERRIDES = ACCUMULATING_OVERRIDES + tuple(
+    f"{attr}_{suffix}" for attr, suffix in itertools.product(ACCUMULATING_OVERRIDES, MODIFIERS)
+)
+
+
+def build(key: str, data: Any) -> Callable:
```

Comment:
Could we have an `Args` section in the docstring describing what `data` is? The type is `Any` and the name is fairly generic so it's hard to understand what `data` actually is.